src/Components/Components/src/Routing/RouteTableFactory.cs

```diff
@@ -83,35 +83,40 @@ static void GetRouteableComponents(List<Type> routeableComponents, Assembly asse
 
     internal static RouteTable Create(List<Type> componentTypes, IServiceProvider serviceProvider)
     {
-        var templatesByHandler = new Dictionary<Type, string[]>();
+        var templatesByHandler = new Dictionary<Type, IReadOnlyList<string>>();
         foreach (var componentType in componentTypes)
         {
             // We're deliberately using inherit = false here.
             //
             // RouteAttribute is defined as non-inherited, because inheriting a route attribute always causes an
             // ambiguity. You end up with two components (base class and derived class) with the same route.
-            var templates = GetTemplates(componentType);
-
-            templatesByHandler.Add(componentType, templates);
+            // We exclude static routes because this is the interactive router.
+            var templates = GetTemplates(componentType, includeStaticRoutes: false);
+            if (templates.Count > 0)
+            {
+                templatesByHandler.Add(componentType, templates);
+            }
         }
         return Create(templatesByHandler, serviceProvider);
     }
 
-    private static string[] GetTemplates(Type componentType)
+    private static IReadOnlyList<string> GetTemplates(Type componentType, bool includeStaticRoutes)
```

Comment:
Replacing `string[]` with `IReadOnlyList<string>` is a possibly-excessive optimization so that we can just build the list in a single pass and don't have to then `.ToArray()` on it. TBH I could have skipped this change but it's not really difficult and I don't think there's any drawback now it's done.